### ISSUE TITLE
[1.3.3] BUILD_NUMBER env var does not enable color in Jenkins

### ISSUE THREAD

**Issue by anthonyemerson** [sbt/sbt#5269]
## steps
sbt version: 1.3.3
scala version: 2.13.1
Jenkins version: 2.150
When running a suite of scala tests using SBT and Jenkins

## problem
There is no ANSI color output in my Jenkins logs

## expectation
As noted in https://www.scala-sbt.org/1.x/docs/Command-Line-Reference.html the sbt.ci flag defaults to false UNLESS the BUILD_NUMBER env variable is set. As Jenkins sets the BUILD_NUMBER env variable I would expect that the ci flag is automatically set to true and the Jenkins logs are properly colored.

**Comment by anthonyemerson** [comment]
See related tickets ##4879 and #4284

**Comment by francisdb** [comment]
This was introduced here https://github.com/sbt/sbt/commit/81481a8af59b749dacc8360a1c62d5de119e9cf4

**Comment by francisdb** [comment]
would change the title to `BUILD_NUMBER env var does not enable color in Jenkins`

**Comment by francisdb** [comment]
The `-Dsbt.ci=true` option seems to not touch color. Locally I have color with `-Dsbt.ci=true` but on Jenkins not so your first description is correct. With `-Dsbt.color=always` on Jenkins supershell is disabled and I have color so that leads me to think the env var is correctly detected.

**Comment by francisdb** [comment]
@eatkins would you mind clarifying what the expected behavior should be?

**Comment by eatkins** [comment]
Honestly I barely remember making this change. I'll defer on this one.

**Comment by eed3si9n** [comment]
I don't think this is related to `-Dsbt.ci`.

In https://github.com/sbt/util/pull/184 I introduced the change so that color is enabled only if stdout console is detected. The purpose was to automatically turn off color when the output is redirected to a file. This change seems to have affected CI.

**Comment by francisdb** [comment]
Only one question needs to be answered here:

**Do we expect `-Dsbt.ci` to influence color and if yes in what way?**

Personally I would like sbt to enable color through `-Dsbt.ci=true` unless color is specifically disabled by `-Dsbt.color=false`. Any proper CI should support color output.

**Comment by anthonyemerson** [comment]
I agree

**Comment by eed3si9n** [comment]
I would amend your proposal so instead of relying on `-Dsbt.ci=true`, it should use default body of `insideCI` setting:

```scala
      insideCI :== sys.env.contains("BUILD_NUMBER") ||
        sys.env.contains("CI") || System.getProperty("sbt.ci", "false") == "true",
```

That should give color back to Jenkins and Travis CI out of box.

**Comment by anthonyemerson** [comment]
That would be preferable

**Comment by francisdb** [comment]
Yes that was the idea, let me come up with a PR

**Comment by eatkins** [comment]
Should be fixed by #5998
